### PR TITLE
Add PLA generator to util

### PR DESCRIPTION
**Pull request by yqszxx**
[PLA](https://en.wikipedia.org/wiki/Programmable_logic_array) generator splitted from `chisel3.experimental.minimizer` (formal `chisel3.experimental.decoder`).

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- new feature/API

#### API Impact
New API: `chisel3.util.experimental.pla`

#### Backend Code Generation Impact
Does this change any generated Verilog?
No, new feature.

#### Desired Merge Strategy

- Squash

#### Release Notes
PLA generator are now in `chisel3.util`.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (3.2.x, 3.3.x, 3.4.0, 3.5.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Review by sequencer — APPROVED** [review]
LGTM
New API name needs to be discussed at dev meeting.

File: src/test/scala/chiselTests/util/experimental/PlaSpec.scala
Comment (sequencer) [inline src/test/scala/chiselTests/util/experimental/PlaSpec.scala:29]: no need to `PrintFullStackTraceAnnotation`?

**Comment by yqszxx** [comment]
Oh my bad, gonna correct this tomorrow 😅

--------------原始邮件--------------
发件人："Jiuyang Liu ***@***.***&gt;;
发送时间：2021年4月4日(星期天) 晚上11:30
收件人："chipsalliance/chisel3" ***@***.***&gt;;
抄送："Boyang Han ***@***.***&gt;;"Author ***@***.***&gt;;
主题：Re: [chipsalliance/chisel3] Add PLA generator to util (#1853)
-----------------------------------

 

 
@sequencer commented on this pull request.
 
 
In src/test/scala/chiselTests/util/experimental/PlaSpec.scala:
 &gt; +        (BitPat("b000"), BitPat("b00000001")), +        (BitPat("b001"), BitPat("b00000010")), +        (BitPat("b010"), BitPat("b00000100")), +        (BitPat("b011"), BitPat("b00001000")), +        (BitPat("b100"), BitPat("b00010000")), +        (BitPat("b101"), BitPat("b00100000")), +        (BitPat("b110"), BitPat("b01000000")), +        (BitPat("b111"), BitPat("b10000000")), +      ) +      table.foreach { case (i, o) =&gt; +        val (plaIn, plaOut) = pla(table) +        plaIn := WireDefault(i.value.U(3.W)) +        chisel3.assert(plaOut === o.value.U(8.W), "Input " + i.toString + " produced incorrect output BitPat(%b)", plaOut) +      } +      stop() +    }, Seq(), Seq(PrintFullStackTraceAnnotation))  
no need to PrintFullStackTraceAnnotation?

**Comment by sequencer** [comment]
Please rebase to master.

**Comment by yqszxx** [comment]
Done

**Comment by sequencer** [comment]
need review from @jackkoenig to this PR, after approve, I'll work on the minimizer part(espresso), and finally land decoder to chisel3

**Comment by sequencer** [comment]
rework to add new inverter layer

**Pull request closed without merging**
